src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -713,12 +713,16 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     ============================================================
                                         _CheckForInvalidConfigurationAndPlatform
 
-    This target checks for errors in statically defined properties.  It always
-    gets executed before any other target.
+    This target checks for errors in statically defined properties.  By setting BeforeTargets, we try
+    to ensure that the target runs before any build related targets.  
+    If your target requires this check and is running as a BeforeTargets of one of the first targets
+    of $(BuildDependsOn), $(RebuildDependsOn), or $(CleanDependsOn) you will need to set your DependsOn
+    to this target.
     ============================================================
     -->
   <Target
-      Name="_CheckForInvalidConfigurationAndPlatform">
+      Name="_CheckForInvalidConfigurationAndPlatform"
+      BeforeTargets="$(BuildDependsOn);$(RebuildDependsOn);$(CleanDependsOn)">
```

Comment:
@cdmihai had suggested also adding `;Build;Rebuild;Clean` to the list just in case someone cleaned out the lists. Seems like a reasonable defense mechanism.